homeassistant/components/nibe_heatpump/__init__.py

```diff
@@ -295,12 +295,12 @@ def _get_coils() -> Iterable[Coil]:
 
         return result
 
-    async def async_shutdown(self):
+    async def async_cleanup_and_shutdown(self) -> None:
         """Make sure a coordinator is shut down as well as it's connection."""
         if self.task:
             self.task.cancel()
             await asyncio.wait((self.task,))
-        self._unschedule_refresh()
+        self.async_shutdown()
```

Comment:
super().async_shutdown() and leave name?

Reply:
Oh, i see you went with non async version. Ok then this is okey.

Reply:
Resolved: now uses async version